internal/utils/lease.go

```diff
@@ -48,17 +49,57 @@ func IsPeerURLTLSEnabledForMembers(ctx context.Context, cl client.Client, logger
 	return tlsEnabledForAllMembers, nil
 }
 
-func parseAndGetTLSEnabledValue(lease coordinationv1.Lease, logger logr.Logger) (bool, error) {
-	if lease.Annotations != nil {
-		if tlsEnabledStr, ok := lease.Annotations[LeaseAnnotationKeyPeerURLTLSEnabled]; ok {
+// isPeerURLTLSDisabledForMembers checks if TLS has been disabled for all existing members of an etcd cluster identified by etcdName and in the provided namespace.
+func isPeerURLTLSDisabledForMembers(ctx context.Context, cl client.Client, logger logr.Logger, etcd *druidv1alpha1.Etcd, replicas int32) (bool, error) {
+	tlsDisabledForAllMembers := true
+	leaseObjMetaSlice, err := ListAllMemberLeaseObjectMeta(ctx, cl, etcd)
+	if err != nil {
+		return false, err
+	}
+	targetMembers := leaseObjMetaSlice[:replicas]
+	for _, leaseObjMeta := range targetMembers {
```

Comment:
removed the variable your suggestion is not correct. In STS component we have to enable the TLS first. This can be for a single member or multi-member and that is governed by replicas passed to this function.